### PR TITLE
Add more validations for FluxConfig

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*

*Description of changes:*
We needed a way to access the FluxConfig spec for validations. We already have a `ParseClusterConfigFromContent` function that accepts any `KindAccessor` type (such as Cluster, FluxConfig, etc.) to parse those contents. So I renamed that function to `ParseConfigFromContent` since it isn't exclusive to the cluster config, and I am using it to pass in the FluxConfig type and parse those contents so that we can have access to that spec is the gitops validations function.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/cluster.go [hunk 1]
```diff
@@ -213,6 +226,19 @@ func GetAndValidateClusterConfig(fileName string) (*Cluster, error) {
 		return nil, err
 	}
 
+	if clusterConfig.Spec.GitOpsRef != nil {
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/cluster.go:229]: you dont validate the fluxConfig here, instead u can do it similar to https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/gitops.go

`GetAndValidateClusterConfig` only validate cluster object itself (including the refs)

File: pkg/api/v1alpha1/cluster.go [hunk 2]
```diff
@@ -582,9 +608,22 @@ func validateGitOps(clusterConfig *Cluster) error {
 	gitOpsRefKind := gitOpsRef.Kind
 	fluxConfigActive := features.IsActive(features.GenericGitProviderSupport())
 
-	if gitOpsRefKind == FluxConfigKind && !fluxConfigActive {
-		return fmt.Errorf("FluxConfig and the generic git provider are not currently supported; " +
-			"to use this experimental feature, please set the environment variable GENERIC_GIT_PROVIDER_SUPPORT to true")
+	if gitOpsRefKind == FluxConfigKind {
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/cluster.go:611]: here too, validateGitOps is to validate cluster obj's gitops config, not the fluxconfig itself

**Review by jiayiwang7 — COMMENTED** [review]
can we also add unit tests?

File: pkg/api/v1alpha1/cluster.go [hunk 3]
```diff
@@ -194,13 +195,25 @@ func GetClusterConfig(fileName string) (*Cluster, error) {
...
 		return clusterConfig, err
 	}
 	return clusterConfig, nil
 }
 
+// GetFluxConfig parses a Cluster object from a multiobject yaml file in disk
+// and sets defaults if necessary
+func GetFluxConfig(fileName string) (*FluxConfig, error) {
+	fluxConfig := &FluxConfig{}
+	err := ParseClusterConfig(fileName, fluxConfig)
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster.go:209]: Should we change this name too then?

File: pkg/api/v1alpha1/cluster.go (hunk 1 above)
Comment (vivek-koppuru) [reply pkg/api/v1alpha1/cluster.go:229]: Agreed, here we only validate if the ref has all the values or something from the perspective of the cluster object.

File: pkg/api/v1alpha1/cluster.go [hunk 4]
```diff
@@ -582,9 +608,22 @@ func validateGitOps(clusterConfig *Cluster) error {
 	gitOpsRefKind := gitOpsRef.Kind
 	fluxConfigActive := features.IsActive(features.GenericGitProviderSupport())
 
-	if gitOpsRefKind == FluxConfigKind && !fluxConfigActive {
-		return fmt.Errorf("FluxConfig and the generic git provider are not currently supported; " +
-			"to use this experimental feature, please set the environment variable GENERIC_GIT_PROVIDER_SUPPORT to true")
+	if gitOpsRefKind == FluxConfigKind {
+		if !fluxConfigActive {
+			return fmt.Errorf("FluxConfig and the generic git provider are not currently supported; " +
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster.go:613]: Are we sticking to the name being "generic git provider" in our documentation too? I like just calling this "git provider" tbh

File: pkg/api/v1alpha1/cluster.go [hunk 5]
```diff
@@ -582,9 +608,22 @@ func validateGitOps(clusterConfig *Cluster) error {
...
+				"to use this experimental feature, please set the environment variable GENERIC_GIT_PROVIDER_SUPPORT to true")
+		}
+		if fluxConfig.Spec.Git != nil {
+			if fluxConfig.Spec.Github != nil {
+				return errors.New("only one provider can be specified in FluxConfig. Please only specify either git or github")
+			}
+			if privateKeyFile, ok := os.LookupEnv(config.EksaGitPrivateKeyTokenEnv); !ok || len(privateKeyFile) <= 0 {
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster.go:620]: Do we need any other validations on the values of these env vars?

Comment (danbudris) [reply pkg/api/v1alpha1/cluster.go:620]: ideally we should do a check on the permissions of the private key file, it should be set at 600

File: pkg/api/v1alpha1/cluster.go (hunk 1 above)
Comment (taneyland) [reply pkg/api/v1alpha1/cluster.go:229]: For sure. Once #2034 merges, I'll refactor this PR to add the validations in the proper place

**Review by danbudris — APPROVED** [review]

**Pull request merged**